packages/grafana-ui/src/components/PanelChrome/PanelContext.ts

```diff
@@ -82,6 +84,17 @@ export interface PanelContext {
    * Called when a panel is changing the sort order of the legends.
    */
   onToggleLegendSort?: (sortBy: string) => void;
+
+  /**
+   * Some panels could be interested in updating the queries.
+   * This optional and not always available in all contexts.
+   */
+  onUpdateQueries?: (queries: DataQuery[]) => void;
+
+  /**
+   * Optional, only some contexts support this. This can fail / be cancelled. Which is why it returns a promise.
+   */
```

Comment:
And this one is odd given that can only translate to a single query type/ so not sure the helper wrapper is a good approach

Reply:
@ryantxu what do you mean, only translate to single query? 

We send a DataFrame array (via panel data ) so the update is an array. How the context updates the data is up to the context